### ISSUE TITLE
ATA doesn't work on Nightlies during the RC period

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#34818]
```sh
mkdir watwatwat
cd watwatwat
npx express-generator
```

**Expected:** ATA works
**Actual:** Everything appears as `any`. `ti-40592.log` contains:

```
[16:27:5.492] Sending response:
    {"kind":"event::beginInstallTypes","eventId":1,"typingsInstallerVersion":"3.8.0-dev.20191026","projectName":"/dev/null/inferredProject1*"}
[16:27:5.493] Response has been sent.
[16:27:5.494] #1 with arguments'["@types/cookie-parser@ts3.8","@types/debug@ts3.8","@types/express@ts3.8","@types/http-errors@ts3.8","@types/jade@ts3.8","@types/morgan@ts3.8","@types/node@ts3.8"]'.
[16:27:5.494] Exec: npm install --ignore-scripts @types/cookie-parser@ts3.8 @types/debug@ts3.8 @types/express@ts3.8 @types/http-errors@ts3.8 @types/jade@ts3.8 @types/morgan@ts3.8 @types/node@ts3.8 --save-dev --user-agent="typesInstaller/3.8.0-dev.20191026"
[16:27:8.295]     Failed. stdout:
    
    stderr:
    npm ERR! code ETARGET
    npm ERR! notarget No matching version found for @types/jade@ts3.8
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Daniel\AppData\Roaming\npm-cache\_logs\2019-10-29T23_27_08_255Z-debug.log
```

**Comment by DanielRosenwasser** [comment]
Ben mentioned that this potentially shouldn't even be happening because types-registry shouldn't have whitelisted `@types/jade`?

**Comment by DanielRosenwasser** [comment]
Looks like we need to change our process so that https://github.com/microsoft/definitelytyped-header-parser/pull/26 is done with the RCs.

**Comment by sandersn** [comment]
Well, sort of. Using only the tools we have today, the process would have to be:

## RC

1. Add 3.8 to definitelytyped-header-parser
2. Update dt-retag's dependency on it.
3. Run dt-retag to add ts3.8 to the tags.

Don't update anybody else's dependency, or it will break DT!

## Release

1. Update dtslint and types-publisher's dependency on definitelytyped-header-parser.
2. Run dt-retag again to add the ts3.8 tag to packages published in the RC period.

I just did the first half, and will do the second half at release.

## Next Time

This is complicated! It would be better to make definitelytyped-header-parser consult npm to see which versions of typescript have actually been published. Then it could offer two lists:

1. Versions which have been published without a suffix -- that is, generally available versions like 3.7 or 3.8.
2. Versions which have been published with or without a suffix like -beta, -dev or -rc.

Then dt-retag would always use list (2) while types-publisher would use list (1) for CI. This would reduce the release tasks to two steps:

### RC
1. Run dt-retag to add the ts3.9 tag.

### Rlease
1. Run dt-retag to add ts3.9 to the packages published in the RC period.

**Comment by sandersn** [comment]
I'm going to wait until I've updated definitelytyped-header-parser skip 2.0-2.7 since that complexity will need to be handled by whatever code I write to fix this bug.
